src/lib.rs

```diff
@@ -231,11 +254,43 @@ impl hal::blocking::spi::Transfer<u8> for Spidev {
 impl hal::blocking::spi::Write<u8> for Spidev {
     type Error = io::Error;
 
-    fn write(&mut self, buffer: &[u8]) -> io::Result<()> {
+    fn try_write(&mut self, buffer: &[u8]) -> io::Result<()> {
         self.0.write_all(buffer)
     }
 }
 
+pub use hal::blocking::spi::{Operation as SpiOperation};
+
+/// Transactional implementation batches SPI operations into a single transaction
+impl hal::blocking::spi::Transactional<u8> for Spidev {
+    type Error = io::Error;
+
+    fn try_exec<'a>(&mut self, operations: &mut [SpiOperation<'a, u8>]) -> Result<(), Self::Error> {
+
+        // Map types from generic to linux objects
+        let mut messages: Vec<_> = operations.iter_mut().map(|a| {
+            match a {
+                SpiOperation::Write(w) => SpidevTransfer::write(w),
+                SpiOperation::Transfer(r) => {
+                    // Clone read to write pointer
+                    // SPIdev is okay with having w == r but this is tricky to achieve in safe rust
+                    let w = unsafe {
+                        let p = r.as_ptr();
+                        std::slice::from_raw_parts(p, r.len())
+                    };
+
+                    SpidevTransfer::read_write(w, r)
+                },
+            }
+        }).collect();
+
+        // Execute transfer
+        self.0.transfer_multiple(&mut messages)?;
+
+        Ok(())
```

Comment:
```suggestion
        self.0.transfer_multiple(&mut messages)
```
Small simplification